android/app/src/main/java/com/mattermost/rnbeta/MattermostManagedModule.java

```diff
@@ -1,27 +1,88 @@
 package com.mattermost.rnbeta;
 
 import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.ParcelFileDescriptor;
+import android.webkit.MimeTypeMap;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
+import androidx.core.content.FileProvider;
 
+import com.facebook.react.bridge.ActivityEventListener;
 import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.BaseActivityEventListener;
+import com.facebook.react.bridge.GuardedResultAsyncTask;
 import com.facebook.react.bridge.Promise;
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReactContextBaseJavaModule;
 import com.facebook.react.bridge.ReactMethod;
 import com.facebook.react.bridge.WritableMap;
 import com.facebook.react.modules.core.DeviceEventManagerModule;
+import com.mattermost.helpers.RealPathUtil;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import java.nio.channels.FileChannel;
 
 public class MattermostManagedModule extends ReactContextBaseJavaModule {
+    private static final String SAVE_EVENT = "MattermostManagedSaveFile";
+    private static final Integer SAVE_REQUEST = 38641;
     private static MattermostManagedModule instance;
     private ReactApplicationContext reactContext;
 
+    private Promise mPickerPromise;
+    private String fileContent;
+
     private static final String TAG = MattermostManagedModule.class.getSimpleName();
 
     private MattermostManagedModule(ReactApplicationContext reactContext) {
         super(reactContext);
         this.reactContext = reactContext;
+        // Let the document provider know you're done by closing the stream.
+        ActivityEventListener mActivityEventListener = new BaseActivityEventListener() {
+            @Override
+            public void onActivityResult(Activity activity, int requestCode, int resultCode, Intent intent) {
+                if (requestCode == SAVE_REQUEST) {
+                    if (mPickerPromise != null) {
+                        if (resultCode == Activity.RESULT_CANCELED) {
+                            mPickerPromise.reject(SAVE_EVENT, "Save operation cancelled");
+                            mPickerPromise = null;
+                        } else if (resultCode == Activity.RESULT_OK) {
```

Comment:
If we can only get Activity.RESULT_OK and RESULT_CANCEL, this could be just else. If that were the case, we can move the mPickerPromise = null to the end of the parent if block.

If that is not the case, we can move the mPickerPromise = null to the end of this if block.